control/control.go

```diff
@@ -814,7 +848,7 @@ func (p *pluginControl) FetchMetrics(ns []string, version int) ([]core.Cataloged
 }
 
 func (p *pluginControl) GetMetric(ns []string, ver int) (core.CatalogedMetric, error) {
-	return p.metricCatalog.Get(ns, ver)
+	return p.metricCatalog.GetMetric(ns, ver)
 }
 
 func (p *pluginControl) GetMetricVersions(ns []string) ([]core.CatalogedMetric, error) {
```

Comment:
After thinking about it I agree, it would be better as a separate issue. I opened #810 yesterday that is related.

My thought was just to move handling of dynamic queries out of the API. Currently the API handles queries with a trailing '*'  [here](https://github.com/intelsdi-x/snap/blob/master/mgmt/rest/metric.go#L43) which returns everything that starts with the same concrete namespace that is given.